docs/guides/testing.md

```diff
@@ -1,105 +1,202 @@
 
-// when creating a store, we get its initial state, create a reset function and add it in the set
-export const create = (createState) => {
-  const store = actualCreate(createState)
-  const initialState = store.getState()
-  storeResetFns.add(() => store.setState(initialState, true))
-  return store
-}
+### UI and Network Testing Tools
 
-// Reset all stores after each test run
-beforeEach(() => {
-  act(() => storeResetFns.forEach((resetFn) => resetFn()))
-})
-```
+#### Web
+
+**We recommend using [React Testing Library (RTL)](https://testing-library.com/docs/react-testing-library/intro)
+to test React components that connect to Zustand**. `RTL` is a simple and complete React DOM
+testing utility that encourages good testing practices. It uses `ReactDOM`'s render function and
+act from `react-dom/tests-utils`. (The [Testing Library](https://testing-library.com/) family of
+tools also includes adapters for many other popular frameworks too.)
+
+#### Native
+
+**We recommend using [Native Testing Library (RNTL)](https://testing-library.com/docs/react-native-testing-library/intro)
+to test React Native components that connect to Zustand**. `RTL` is a simple and complete native
+testing utility that encourages good testing practices. The API is the same as `RTL` library with
+some mire differences like `RNTL` queries are implemented independently, unlike other wrappers that
+use `DOM Testing Library`.
 
-The way you mock a dependency depends on your test runner/library.
+#### Network Requests
 
-In [jest](https://jestjs.io/), you can create a `__mocks__/zustand.js` and place the code in that
-file. If your app is using `zustand/vanilla` instead of `zustand`, then you'll have to place the
-above code in `__mocks__/zustand/vanilla.js`.
+We also recommend using [Mock Service Worker (MSW)](https://mswjs.io/) to mock network requests, as
+this means your application logic does not need to be changed or mocked when writing tests.
 
-### TypeScript usage
+- **React Testing Library**
+  - [DOM Testing Library: Setup](https://testing-library.com/docs/dom-testing-library/setup)
+  - [React Testing Library: Setup](https://testing-library.com/docs/react-testing-library/setup)
+  - [Testing Library Jest-DOM Matchers](https://testing-library.com/docs/ecosystem-jest-dom)
+- **Native Testing Library**
+  - [Native Testing Library: Setup](https://testing-library.com/docs/react-native-testing-library/setup)
+- **Mock Service Worker**
+  - [MSW: Installation](https://mswjs.io/docs/getting-started/install)
+  - [MSW: Setting up mock requests](https://mswjs.io/docs/getting-started/mocks/rest-api)
+  - [MSW: Mock server configuration for Node](https://mswjs.io/docs/getting-started/integrate/node)
 
-If you are using zustand, as documented in [TypeScript Guide](./typescript.md), use the following
-code:
+## Setting Up Zustand for testing
+
+Since `Jest` and `Vitest` have slight difference like `Vitest` uses **ES modules** and `Jest`
```

Comment:
> Since `Jest` and `Vitest` have slight difference like `Vitest` uses **ES modules** and `Jest` uses **CommonJS modules**

Maybe

> Since `Jest` and `Vitest` have slight differences, like `Vitest` using **ES modules** and `Jest` using **CommonJS modules**,

(tense, "s" and an additional ",")